backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/graphicpromotion/internal/BpkGraphicPromoImpl.kt

```diff
@@ -0,0 +1,282 @@
+
+@Composable
+internal fun BpkGraphicPromoImpl(
+    headline: String,
+    image: @Composable BoxScope.() -> Unit,
+    modifier: Modifier = Modifier,
+    kicker: String? = null,
+    subHeadline: String? = null,
+    overlayType: BpkOverlayType? = null,
+    bpkGraphicPromoVariant: BpkGraphicPromoVariant = BpkGraphicPromoVariant.OnDark,
+    bpkGraphicPromoVerticalAlignment: BpkGraphicPromoVerticalAlignment = BpkGraphicPromoVerticalAlignment.Top,
+    sponsor: Sponsor? = null,
+    sponsorLogo: (@Composable () -> Unit)? = null,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    tapAction: () -> Unit = {},
+) {
+    val roundedCornerShape = RoundedCornerShape(BpkBorderRadius.Md)
+    val contentDescription = listOfNotNull(kicker, headline, subHeadline, sponsor?.accessibilityLabel)
+        .joinToString(separator = ", ")
+    Box(
+        modifier = modifier
+            .aspectRatio(RATIO_PORTRAIT)
+            .clip(roundedCornerShape)
+            .clickable(
+                interactionSource = interactionSource,
+                indication = null,
+                onClick = tapAction,
+            )
+            .semantics(mergeDescendants = true) {
+                role = Role.Button
+                this.contentDescription = contentDescription
+            },
+    ) {
+        if (overlayType != null) {
+            BpkOverlay(
+                modifier = Modifier
+                    .matchParentSize()
+                    .indication(interactionSource, InteractiveBackgroundIndication),
+                overlayType = overlayType,
+                foregroundContent = {
+                    ForegroundContent(
+                        headline = headline,
+                        kicker = kicker,
+                        subHeadline = subHeadline,
+                        bpkGraphicPromoVariant = bpkGraphicPromoVariant,
+                        bpkGraphicPromoVerticalAlignment = bpkGraphicPromoVerticalAlignment,
+                        sponsor = sponsor,
+                        sponsorLogo = sponsorLogo,
+                    )
+                },
+                content = image,
+            )
+        } else {
+            Box(
+                modifier = Modifier
+                    .matchParentSize()
+                    .indication(interactionSource, InteractiveBackgroundIndication),
+                content = image,
+            )
+            ForegroundContent(
+                headline = headline,
+                kicker = kicker,
+                subHeadline = subHeadline,
+                bpkGraphicPromoVariant = bpkGraphicPromoVariant,
+                bpkGraphicPromoVerticalAlignment = bpkGraphicPromoVerticalAlignment,
+                sponsor = sponsor,
+                sponsorLogo = sponsorLogo,
+            )
+        }
+    }
+}
+
+@Composable
+private fun SponsorOverlayView(
+    textColor: Color,
+    sponsor: Sponsor?,
```

Comment:
We need to have modifiers in all composables. Also, is there a specific reason why content might be null?